dsa/DSA || GAS STATION(GREEDY) #33.py

```diff
@@ -0,0 +1,20 @@
+def gascheck(j,counter,remaining_gas,cost_for_next_station):
+        if(counter==n):
+            return 106
+        elif(remaining_gas>=cost_for_next_station):
+            j=j+1
+            if(j>=n):
+                j=j-n
+            remaining_gas=gas[j]+remaining_gas-cost_for_next_station
+            cost_for_next_station=cost[j]
+            counter=counter+1
+            gascheck(j,counter,remaining_gas,cost_for_next_station)
+gas=list(map(int,input().split(",")))
+cost=list(map(int,input().split(",")))
+n=len(gas)
+index=-1
+for i in range(n):
+    if(gascheck(i,0,gas[i],cost[i])==106):
```

Comment:
Your function returns 106 only when counter becomes n, since in the else part u r not returning anything so by default the function returns `None`, that's why this if condition will never be true as the value returned by the function is always `None`

Reply:
> I have pointed out the mistake in your code, also don't push the chages for another issue in this you'll have to remove the added file from this pr, u will need to make a separate pr for the other issue by creating a branch, then adding your changes to that branch and then making a pr from that branch

Yes sir I was trying to do the same but messed up my steps. It was done by mistake  .Will revert the commit once I figure out how to do it.